src/utils/pgwire/src/pg_extended.rs

```diff
@@ -123,16 +123,33 @@ impl PgStatement {
         self.row_description.clone()
     }
 
-    pub fn instance(&self, portal_name: String, params: &[Bytes]) -> PgPortal {
+    pub async fn instance<SM: SessionManager>(
+        &self,
+        session: Arc<SM::Session>,
+        portal_name: String,
+        params: &[Bytes],
+    ) -> Result<PgPortal, ()> {
         let statement = cstr_to_str(&self.query_string).unwrap().to_owned();
 
         if params.is_empty() {
-            return PgPortal {
+            let row_description =
+                if statement.starts_with("select") || statement.starts_with("SELECT") {
+                    if let Ok(rows) = session.infer_return_type(statement.as_str()).await {
+                        rows
+                    } else {
+                        return Err(());
```

Comment:
What about use `?`

Reply:
> What about use `?`

infer_return_type -> Result<-, BoxedError>
We can't covert Result<-, BoxedError> to Result<-,()> when using '?'. Or is there another way can process it more concisely?

Reply:
try .map_err?